system/Database/BaseConnection.php

```diff
@@ -45,6 +45,10 @@
  * @property int        $transDepth
  * @property bool       $transFailure
  * @property bool       $transStatus
+ *
+ * @template TConnection of bool|object|resource
```

Comment:
You're going to need to explain the `bool` to me... Is this to support returning `false` on failure?

Reply:
On second look, I think this should have been `false` not `bool`.